src/components/UriViewer/LogViewer.ts

```diff
@@ -0,0 +1,224 @@
+      fetch(uri);
+    }
+  }, [logs, currentlyActiveLog, fetchLogContent]);
+
+  const renderActiveTextContent = () => {
+    if (activeTextContent === undefined) {
+      return div([centeredSpinner()]);
+    }
+    if (activeTextContent === null) {
+      return div([
+        "Log file could not be loaded. If the workflow or task is still in progress, the log file likely hasn't been generated yet. Some logs may be unavailable if the workflow or task failed before they could be generated.",
+      ]);
+    }
+    return div([activeTextContent]);
+  };
+
+  const tabsArray: SimpleTabProps[] = logs.map((log) => {
+    return { key: log.logKey, title: log.logTitle, width: tabMaxWidth } as SimpleTabProps;
+  });
+
+  return h(
+    Modal,
+    {
+      onDismiss,
+      title: modalTitle,
+      titleChildren: h(
+        InfoBox,
+        {
+          style: { marginLeft: '1ch' },
+          tooltip: undefined,
+          size: undefined,
+          side: undefined,
+          iconOverride: undefined,
+        },
+        [
+          div({ style: { fontWeight: 'bold' } }, ['Execution:']),
+          div({ style: { marginLeft: '1rem', marginBottom: '1rem' } }, [
+            'Each workflow has a single execution log which comes from the engine running your workflow. Errors in this log might indicate a Terra systems issue, or a problem parsing your WDL.',
+          ]),
+          div({ style: { fontWeight: 'bold' } }, ['Task Standard Out/Error:']),
+          div({ style: { marginLeft: '1rem' } }, [
+            "Task logs are from user-defined commands in your WDL. You might see an error in these logs if there was a logic or syntax error in a command, or if something went with the tool you're running.",
+          ]),
+          div({ style: { marginTop: '1rem', fontWeight: 'bold' } }, ['Backend Standard Out/Error:']),
+          div({ style: { marginLeft: '1rem' } }, [
+            "Backend logs are from the Azure Cloud compute job that prepares your task to run and cleans up after. You might see errors in these logs if the there was a problem downloading the task's input files or pulling its container, or if something went wrong on the compute node while the task was running.",
+          ]),
+        ]
+      ),
+      showCancel: false,
+      showX: true,
+      showButtons: false,
+      width: modalMaxWidth,
+    },
+    [
+      h(
+        SimpleTabBar,
+        {
+          value: currentlyActiveLog?.logKey,
+          'aria-label': 'Log file tabs',
+          onChange: (key: string) => {
+            const newLog = logs.find((log) => log.logKey === key);
+            setCurrentlyActiveLog(newLog);
+          },
+          tabs: tabsArray,
+        },
+        []
+      ),
+      div(
+        {
+          style: {
+            display: 'flex',
+            justifyContent: 'space-between',
+            alignItems: 'center',
+            paddingTop: '0.5rem',
+          },
+        },
+        [
+          span({}, [
```

Comment:
```suggestion
          span([
```

Empty props objects can be omitted.